storagenode/orders/service.go

```diff
@@ -71,102 +71,135 @@ type DB interface {
 	ListUnsentBySatellite(ctx context.Context) (map[storj.NodeID][]*Info, error)
 
 	// Archive marks order as being handled.
-	Archive(ctx context.Context, requests ...ArchiveRequest) error
+	Archive(ctx context.Context, archivedAt time.Time, requests ...ArchiveRequest) error
 	// ListArchived returns orders that have been sent.
 	ListArchived(ctx context.Context, limit int) ([]*ArchivedInfo, error)
 	// CleanArchive deletes all entries older than ttl
 	CleanArchive(ctx context.Context, ttl time.Duration) (int, error)
 }
 
-// SenderConfig defines configuration for sending orders.
-type SenderConfig struct {
-	Interval time.Duration `help:"duration between sending" default:"1h0m0s"`
-	Timeout  time.Duration `help:"timeout for sending" default:"1h0m0s"`
+// Config defines configuration for sending orders.
+type Config struct {
+	SenderInterval time.Duration `help:"duration between sending" default:"1h0m0s"`
+	ArchiveTTL     time.Duration `help:"length of time to archive orders before deletion" default:"1080h0m0s"` // 45 days
+	Timeout        time.Duration `help:"timeout for sending" default:"1h0m0s"`
 }
 
-// Sender sends every interval unsent orders to the satellite.
-type Sender struct {
+// Service sends every interval unsent orders to the satellite.
+type Service struct {
 	log    *zap.Logger
-	config SenderConfig
+	config Config
 
 	transport transport.Client
 	orders    DB
 	trust     *trust.Pool
 
-	Loop sync2.Cycle
+	Sender  sync2.Cycle
+	Cleanup sync2.Cycle
 }
 
-// NewSender creates an order sender.
-func NewSender(log *zap.Logger, transport transport.Client, orders DB, trust *trust.Pool, config SenderConfig) *Sender {
-	return &Sender{
+// NewService creates an order service.
+func NewService(log *zap.Logger, transport transport.Client, orders DB, trust *trust.Pool, config Config) *Service {
+	return &Service{
 		log:       log,
 		transport: transport,
 		orders:    orders,
 		config:    config,
 		trust:     trust,
 
-		Loop: *sync2.NewCycle(config.Interval),
+		Sender:  *sync2.NewCycle(config.SenderInterval),
+		Cleanup: *sync2.NewCycle(24 * time.Hour),
 	}
 }
 
 // Run sends orders on every interval to the appropriate satellites.
-func (sender *Sender) Run(ctx context.Context) (err error) {
+func (service *Service) Run(ctx context.Context) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	return sender.Loop.Run(ctx, sender.runOnce)
+
+	var group errgroup.Group
+	service.Sender.Start(ctx, &group, func(ctx context.Context) error {
+		err := service.sendOrders(ctx)
+		if err != nil {
+			service.log.Error("error sending orders: ", zap.Error(err))
+		}
+		return nil
+	})
+	service.Cleanup.Start(ctx, &group, func(ctx context.Context) error {
```

Comment:
You could pass `service.cleanArchive` as argument and handle error logging in `cleanArchive`. Same for `sendOrders`